packages/server/src/server.ts

```diff
@@ -202,17 +218,38 @@ export class Server extends EventEmitter {
     // Invoke the handler for the method requested
     const handler = this.handlers[req.method as keyof Handlers]
     if (handler) {
-      return handler.send(req, res).catch(onError)
+      return handler.send(req, res, context).catch(onError)
     }
 
-    return this.write(res, 404, 'Not found\n')
+    return this.write(req, res, 404, 'Not found\n', {}, context)
   }
 
-  write(res: http.ServerResponse, status: number, body = '', headers = {}) {
+  write(
+    req: http.IncomingMessage,
+    res: http.ServerResponse,
+    status: number,
+    body = '',
+    headers = {},
+    context: CancellationContext
+  ) {
+    const isAborted = context.signal.aborted
+
     if (status !== 204) {
       // @ts-expect-error not explicitly typed but possible
       headers['Content-Length'] = Buffer.byteLength(body, 'utf8')
     }
+
+    if (isAborted) {
+      // @ts-expect-error not explicitly typed but possible
+      headers['Connection'] = 'close'
+      res.writeHead(status, headers)
+      res.write(body)
+      const sent = res.end()
+      req.destroy()
```

Comment:
not sure if it matters but `res.end()` has a callback, shouldn't we call `destroy` there to avoid a race condition?

Reply:
Yes, good point

Reply:
i'm now calling `req.destroy()` on the `res.on('finish')` event which is essentially the same thing as the callback